vscode/microsoft-kiota/src/extension.ts

```diff
@@ -60,21 +57,53 @@ export async function activate(
   await loadTreeView(context);
   await checkForLockFileAndPrompt(context);
   let codeLensProvider = new CodeLensProvider();
+  let deepLinkParams: Record<string, string|undefined> = {};
   context.subscriptions.push(
     vscode.window.registerUriHandler({
       handleUri: async (uri: vscode.Uri) => {
         if (uri.path === "/") {
           return;
         }
+        console.log(uri);
         const queryParameters = getQueryParameters(uri);
+        console.log(queryParameters);
         if (uri.path.toLowerCase() === "/opendescription") {
           reporter.sendTelemetryEvent("DeepLink.OpenDescription");
           const descriptionUrl = queryParameters["descriptionurl"];
+          const name = getSanitizedString(queryParameters["name"]);
+          const source = getSanitizedString(queryParameters["source"]);
+          reporter.sendTelemetryEvent("DeepLink.OpenDescription", {"source": source}); 
+          let lowercasedKind: string = queryParameters["kind"]?.toLowerCase();
+          let validKind: string | undefined = ["plugin", "client"].indexOf(lowercasedKind) > -1 ? lowercasedKind : undefined ;
+          if (!validKind){
+            vscode.window.showInformationMessage("Invalid parameter 'kind' deeplinked.");
```

Comment:
Can we also log the actual value and expected/valid values? Useful for diagnostics.

Reply:
It may make the error message too bulky given the small informational dialog box at the bottom of the screen. Maybe we can add check logs for details.